data/features/picture_in_picture.toml

```diff
@@ -0,0 +1,12 @@
+name = "Picture-in-Picture"
```

Comment:
So should this have a `slug = "picture_in_picture"` entry?

Reply:
If the features are defined in separate documents, I think it makes the most sense for the slugs to match the filenames, so a couple of options are:
- Adopt a convention that slugs are inferred from filenames at the time of deserialization, or
- Require that slugs are defined explicitly and match the filename

I prefer the economy of the former but wouldn't object too strenuously to the latter. :)

Reply:
If you take a look at #43, I have included the ability [to add metadata](https://github.com/mozilla/nimbus-shared/pull/43/files#diff-ad749c700de8779dfbde63f6463d8a43R1-R2) for data defined in JSON or TOML files. How would you feel about a metadata value like `useFileForSlug: true`, that would then pull the slug from the filename?

The reason I don't want to do this in general is that some types of data won't have slugs, or will have other identifying paths for them. I'd like to avoid making something specific to feature definitions.

Reply:
I think auto generating the slugs from the filename will make it harder to correlate data later, ie if someone is seeing a slug in some other system and comes to this repo to look/search for it, they might not know that it's inferred from the filename.  Plus tying it to the filename creates an interepedency on the two so that if you wanted/needed to change one and not the other you'd be locked in.  In other schemas, ie for audience, the slug is explicitly declared, so again I'd suggest that consistency between representations will make it easier to reason about the contents of this repo.

Reply:
The useFilenameForSlug convention works for me; thanks!

> Plus tying it to the filename creates an interepedency on the two so that if you wanted/needed to change one and not the other you'd be locked in.

Yeah, I'm asserting that would be invalid. This also naturally requires that slugs are unique, which is pleasant. I take your point that filenames don't grep as well as flat strings; it trades off a little cost for writers vs. readers. I'd normally optimize for readers but I think we'll have inexpert writers adding these pretty frequently.

> In other schemas, ie for audience, the slug is explicitly declared

Well, the slug isn't a member of the Audience object at all. Should it be?